services/searcher.ts

```diff
@@ -458,6 +459,60 @@ class Searcher {
     };
   }
 
+  filterSection(
+    curSection: Section,
+    honorsFilter: boolean,
+    campusFilter: string
+  ): boolean {
+    if (honorsFilter && !curSection.honors) {
+      return false;
+    }
+    if (campusFilter !== "" && curSection.campus !== campusFilter) {
+      return false;
+    }
+
+    return true;
+  }
+
+  filterResults(filters, results): SearchResult[] {
+    // Do extra filtering if Honors or Campus Filters are being applied since those filter sections,
+    // which ElasticSearch doesn't actually filter out.
+
+    const validFilters = this.validateFilters(filters);
+
+    const honorsFilter: boolean = validFilters["honors"] ? true : false;
```

Comment:
```suggestion
    const honorsFilter = validFilters["honors"] ?? false;
```

We want to use the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) here.

Using a conditional in the format 
```js
if (condition) {
   return true
} else {
   return false
}
```
is generally a bad design - you could just return the `condition`. However, in this case, the conditional is necessary for Javascript to cast the type to a boolean. We want to avoid that, since that sort of casting can mask errors (eg. if for some reason `"honors": "error"`, the string `"error"` would cast to `true` (since it's non-empty). However, that should break our code, since its a completely unexpected value). 

Note - read the [Examples page for the `??` operator - it's quick](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing#examples)